app/uk/gov/hmrc/nisp/models/NIRecordTaxYear.scala

```diff
@@ -16,17 +16,24 @@
 
 package uk.gov.hmrc.nisp.models
 
+import org.joda.time.LocalDate
 import play.api.libs.json.Json
 import uk.gov.hmrc.nisp.utils.Constants
 
 case class NIRecordTaxYear(taxYear: Int, qualifying: Boolean,classOneContributions: BigDecimal,
                            classTwoCredits: Int, classThreeCredits: Int, otherCredits: Int,
                            classThreePayable: Option[BigDecimal], classThreePayableBy: Option[NpsDate],
                            classThreePayableByPenalty: Option[NpsDate], payable: Boolean, underInvestigation: Boolean) {
+
   val displayableTaxYear: String = s"$taxYear-${(taxYear + 1).toString.substring(Constants.shortYearStartCharacter,Constants.shortYearEndCharacter)}"
-}
 
+  val checkCutOffDate = (localClassThreePayableBy:Option[NpsDate],currentDate:LocalDate) => classThreePayableBy match {
```

Comment:
* Please provide a type on this public method.
* Why is the classThreePayableBy being passed in when it's part of the class? Doesn't that defeat the point of it being on the class?